File: C++/131A. cAPS lOCK.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin>>s;
    int cl = 0, sl = 0;
    for(int i=0; i<s.size(); i++){
        if(isupper(s[i])||(islower(s[i])&&s.size()==1)){
            cl++;
        }
        if(islower(s[i])&&i>0||(isupper(s[i])&&s.size()==1)){
            sl++;
        }
    }
    for(int i=0; i<s.size(); i++){
        if(sl>0&&sl!=s.size()){
            cout<<s[i];
        }
        else if(cl==s.size()&&s.size()>1){
            char ch;
            ch=tolower(s[i]);
                cout<<ch;
        }
        else if(cl>0&&sl==0){
            char ch;
            if(islower(s[i])&&i==0){
                ch=toupper(s[i]);
                cout<<ch;
            }
            else if(isupper(s[i])&&i>0){
                ch=tolower(s[i]);
                cout<<ch;
            }
            else{
                cout<<s[i];
            }
        }
        else{
            char ch;
            ch = tolower(s[i]);
            cout<<ch;
        }
    }
    return 0;
}

File: Python/131A. cAPS lOCK.py
s =  list(input())
i, cl, sl = 0, 0, 0
while i<len(s):
    if s[i].isupper() or (s[i].islower() and len(s) == 1):
        cl += 1
    if (s[i].islower() and i>0) or (s[i].isupper() and len(s) == 1):
        sl += 1
    i += 1
i = 0
while i<len(s):
    if sl > 0 and sl != len(s):
        pass
    elif cl == len(s) and len(s)>1:
        s[i] = s[i].lower()
    elif cl>0 and sl == 0:
        if s[i].islower() and i == 0:
            s[i] = s[i].upper()
        elif s[i].isupper() and i>0:
            s[i] = s[i].lower()
        else:
            pass
    else:
        s[i] = s[i].lower()
    i += 1
print("".join(s))